crates/vm/levm/src/gas_cost.rs

```diff
@@ -637,15 +648,26 @@ pub fn call(
     value_to_transfer: U256,
     gas_from_stack: U256,
     gas_left: u64,
+    spec_id: SpecId,
 ) -> Result<(u64, u64), VMError> {
     let memory_expansion_cost = memory::expansion_cost(new_memory_size, current_memory_size)?;
 
-    let address_access_cost = address_access_cost(
-        address_was_cold,
-        CALL_STATIC,
-        CALL_COLD_DYNAMIC,
-        CALL_WARM_DYNAMIC,
-    )?;
+    // before berlin the cost was 700. This is done in EIP-2929
+    let address_access_cost = if spec_id >= SpecId::BERLIN {
+        address_access_cost(
+            address_was_cold,
+            CALL_STATIC,
+            CALL_COLD_DYNAMIC,
+            CALL_WARM_DYNAMIC,
+        )?
+    } else {
+        address_access_cost(
+            address_was_cold,
+            CALL_STATIC,
+            CALL_COLD_DYNAMIC,
+            CALL_PRE_BERLIN,
+        )?
+    };
```

Comment:
Nit. You can do the `if else` just for the value.
```suggestion
    let address_access_cost = address_access_cost(
        address_was_cold,
        CALL_STATIC,
        CALL_COLD_DYNAMIC,
        if spec_id >= SpecId::BERLIN {
            CALL_WARM_DYNAMIC
        } else {
            CALL_PRE_BERLIN
        },
    )?;
```